src/main/java/appeng/container/implementations/ContainerPatternTerm.java

```diff
@@ -450,6 +459,11 @@ public void detectAndSendChanges()
 				this.craftingMode = this.ct.isCraftingRecipe();
 				this.updateOrderOfOutputSlots();
 			}
+
+			if( this.substitute != this.ct.isSubstitution() )
+			{
+				this.substitute = this.ct.isSubstitution();
+			}
```

Comment:
so its always `this.substitute = this.ct.isSubstitution();`? any performance implications why the `if`is used?